docs/metasploit-framework.wiki/Definition-of-Module-Reliability-Side-Effects-and-Stability.md

```diff
@@ -70,3 +70,4 @@ Example:
 | FIRST_ATTEMPT_FAIL | The module may fail for the first attempt |
 | REPEATABLE_SESSION | The module is expected to get a session every time it runs |
 | UNRELIABLE_SESSION | The module isn't expected to get a shell reliably (such as only once) |
+| EVENT_DEPENDENT    | The module may not open a shell until an external event occurs. For instance, a cron job, machine restart, user interaction within a GUI, etc |
```

Comment:
Discussion:

Alternatively, there was the idea from bcoles to track the potential scenarios separately

This might be _too_ granular, but events that were discussed:

- `USER_INTERACTION_REQUIRED` - i.e. A user performing an action on a GUI - logging in, clicking a particular menu item etc
- `REBOOT_REQUIRED` - Machine reboot required
- `SERVICE_RESTART_REQUIRED` - Requires a reboot of the service
- `SCHEDULED_TASK_REQUIRED` - Depends on a scheduled automated task to complete, i.e. a cron job/windows task

Maybe instead of adding this to the existing reliability section, we add a new bucket for event dependencies? That would make consuming the metadata easier, as automated tools can just bail early if `module.required_events.any?`